From 68939a78ce8954828a2501db0853a744e749255b Mon Sep 17 00:00:00 2001
From: Ryan McCabe <rmccabe@redhat.com>
Date: Fri, 12 Apr 2013 12:10:41 -0400
Subject: [PATCH] modcluster: Fix for cman_get_node() call

cman_get_node() is called in modclusterd without first zeroing out
the node name in the cman_node_t structure passed in to it. This causes
the call to fail. When corosync debugging is enabled, a debugging
message caused by this failure is logged each time the call is made,
which happens at least once per 5 seconds.

This patch zeroes the structure before the call to cman_get_node()

Patch from Christine Caulfield <ccaulfie@redhat.com>

Resolves: rhbz#951470

Signed-off-by: Ryan McCabe <rmccabe@redhat.com>
---
 ricci/modules/cluster/clumon/src/daemon/Monitor.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ricci/modules/cluster/clumon/src/daemon/Monitor.cpp b/ricci/modules/cluster/clumon/src/daemon/Monitor.cpp
index 761fa03..60e2a37 100644
--- a/ricci/modules/cluster/clumon/src/daemon/Monitor.cpp
+++ b/ricci/modules/cluster/clumon/src/daemon/Monitor.cpp
@@ -564,6 +564,7 @@ String Monitor::nodename(const vector<String>& nodenames) {
 	cman_handle_t ch = cman_init(NULL);
 	if (ch != NULL) {
 		cman_node_t this_node;
+		memset(&this_node, 0, sizeof(this_node));
 		if (cman_get_node(ch, CMAN_NODEID_US, &this_node) == 0) {
 			cman_finish(ch);
 			return String(this_node.cn_name);
-- 
1.8.3.1

